ProjFS.Mac/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -83,7 +86,8 @@ kern_return_t VirtualizationRoots_Init()
         return KERN_FAILURE;
     }
     
-    s_maxVirtualizationRoots = 128;
+    // Start with a small size so the resizing logic is regularly tested
+    s_maxVirtualizationRoots = 4;
```

Comment:
If this is initialized to 4, why does the unit test need this code?

```
    // Start with 4 "full" array items, forcing resize on next insertion
    Memory_FreeArray(s_virtualizationRoots, s_maxVirtualizationRoots);
    s_maxVirtualizationRoots = 4;
    s_virtualizationRoots = Memory_AllocArray<VirtualizationRoot>(s_maxVirtualizationRoots);
    memset(s_virtualizationRoots, 0, s_maxVirtualizationRoots * sizeof(s_virtualizationRoots[0]));
```

Is it just to make sure the array has a size of 4 for the test?

Reply:
> Is it just to make sure the array has a size of 4 for the test?

Yes, in case any other tests left the array in some other state. I guess we could init the array during the test's `setUp` but I think that would move that precondition of the test too far apart from the initialisation code, so I'd say in that case we should move the test to a new test class that just contains the tests for the array resizing logic.